### PR TITLE
Upgrade to API version 2019

### PR DESCRIPTION
**Pull request by Namiii**
**TASK**=>[T12812](https://phabricator.omise.co/T12812)

**SUMARRY**
A large PR where we upgraded the API version used in the library from `2017-11-02` to `2019-05-29`. In order to accommodate this change, we also needed to: 

a) Changed variables in a number of Model classes
b) Added new variables to some Model classes
c) Made a number of changes to the sample data we use for testing
d) Fixed broken tests

**DOCS**
Here is an [upgrade guide](https://www.omise.co/upgrade-guide-2017-to-2019) for the new API version

**Note**
The PR is not really as big as it looks, a large number of files that have been touched are sample data files, which can be mainly ignored.

### CODE REVIEW

**Comment by nuxzero** [comment]
Is this PR included new API or not?

**Comment by Namiii** [comment]
@nuxzero that's all it includes

**Comment by nuxzero** [comment]
I can't find Capability API in this PR. 🤔

**Comment by Namiii** [comment]
@nuxzero This is just for the changes in the current API. Capability API will need to be its own PR.

**Comment by nuxzero** [comment]
OK

**Comment by nuxzero** [comment]
One more question, does all live test are passing?

**Comment by Namiii** [comment]
@nuxzero Yes, I checked all, and they should all pass. We can do another run before the release.

**Comment by nuxzero** [comment]
Cool.

File: src/main/java/co/omise/models/Refund.java [hunk 1]
```diff
@@ -169,7 +198,8 @@ protected HttpUrl path() {
 
         @Override
         protected ResponseType<ScopedList<Refund>> type() {
-            return new ResponseType<>(new TypeReference<ScopedList<Refund>>() {});
+            return new ResponseType<>(new TypeReference<ScopedList<Refund>>() {
```
Comment (nuxzero) [inline src/main/java/co/omise/models/Refund.java]: Should this be the same line?

File: src/main/java/co/omise/models/Transaction.java [hunk 2]
```diff
@@ -76,7 +85,7 @@ public GetRequestBuilder(String transactionId) {
         }
 
         @Override
-        protected HttpUrl path() throws IOException {
+        protected HttpUrl path() {
```
Comment (nuxzero) [inline src/main/java/co/omise/models/Transaction.java]: Why do we need to delete the `throws` from method signature?

File: src/main/java/co/omise/models/Refund.java (hunk 1 above)
Comment (Namiii) [reply src/main/java/co/omise/models/Refund.java]: That's how the IDE cleans it up, I've no preference

File: src/main/java/co/omise/models/Transaction.java (hunk 2 above)
Comment (Namiii) [reply src/main/java/co/omise/models/Transaction.java]: They are unneeded, from what I could see. Also got a warning in the IDE.

File: src/main/java/co/omise/models/TransactionDirection.java [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 import com.fasterxml.jackson.annotation.JsonProperty;
 
-public enum TransactionType {
+public enum TransactionDirection {
     @JsonProperty("credit")Credit,
     @JsonProperty("debit")Debit,
```
Comment (pitiphong-p) [inline src/main/java/co/omise/models/TransactionDirection.java]: Nitpicky detail here, but should it have a space between the declaration?

File: src/main/java/co/omise/models/schedules/Occurrence.java [hunk 4]
```diff
@@ -19,9 +19,9 @@
  */
 public class Occurrence extends Model {
     private String schedule;
-    @JsonProperty("schedule_date")
+    @JsonProperty("scheduled_on")
     private DateTime scheduleDate;
```
Comment (pitiphong-p) [inline src/main/java/co/omise/models/schedules/Occurrence.java]: I believe that the type of `on` properties should not be DateTime. I'm not sure what it should be, maybe `LocalDate` from Joda Time.

Comment (Namiii) [reply src/main/java/co/omise/models/schedules/Occurrence.java]: Hmmm true, these have already been here and working. I'll look into what we can replace them with 👍

File: src/main/java/co/omise/models/TransactionDirection.java (hunk 3 above)
Comment (Namiii) [reply src/main/java/co/omise/models/TransactionDirection.java]: Between `)` and the declaration you mean? Yeah correct

Comment (pitiphong-p) [reply src/main/java/co/omise/models/TransactionDirection.java]: Yup

File: src/main/java/co/omise/models/schedules/Occurrence.java (hunk 4 above)
Comment (pitiphong-p) [reply src/main/java/co/omise/models/schedules/Occurrence.java]: OK, so I should pass on this for this PR?

Comment (Namiii) [reply src/main/java/co/omise/models/schedules/Occurrence.java]: No, I'll try and make it work

Comment (Namiii) [reply src/main/java/co/omise/models/schedules/Occurrence.java]: @pitiphong-p Will ping you once it's done to check it again 👌

Comment (Namiii) [reply src/main/java/co/omise/models/schedules/Occurrence.java]: Used `LocalDate` and it worked fine. Will push it soon

**Comment by Namiii** [comment]
@pitiphong-p Done. Please have another look 🙏

**Comment by pitiphong-p** [comment]
OK that's cool. I think you can merge this one :)

**Comment by Namiii** [comment]
@pitiphong-p changes made, please have another look 🙏

**Comment by Namiii** [comment]
@pitiphong-p okk cheers 👌

**Pull request merged**